lib/src/Pos/Launcher/Resource.hs

```diff
@@ -292,9 +292,9 @@ allocateNodeContext ancd txpSettings ekgStore = do
     -- TODO synchronize the NodeContext peers var with whatever system
     -- populates it.
     peersVar <- newTVarIO mempty
-    let slotDuration :: Integer
-        slotDuration = toMicroseconds . bvdSlotDuration $ gdBlockVersionData genesisData
-    ncSubscriptionKeepAliveTimer <- newTimer $ 3 * fromIntegral slotDuration
+    let slotDuration :: Microsecond
+        slotDuration = convertUnit . bvdSlotDuration $ gdBlockVersionData genesisData
+    ncSubscriptionKeepAliveTimer <- newTimer $ 3 * slotDuration
```

Comment:
This is not the same timer that the diffusion layer uses. Fair enough, given that in `develop`, the diffusion layer isn't actually used.

Perhaps this should instead be back-ported on top of https://github.com/input-output-hk/cardano-sl/pull/2154

Reply:
ok, I'll look into this part on Monday morning. I'll push now fixes for the other two changes that you requested though.

Reply:
Thanks.
The place to look is `Pos.Block.Network.Retrieval` and `Pos.Block.Worker`, which pass along a timer and reset it whenever a block comes through. Really though, this should all be done via the diffusion layer. The timer can be created in `fullDiffusionLayer` and passed to the subscription feature, and also to the block listeners, to  get the same result.

Reply:
At the moment merging #2156 brings quite a few conflicts.  I'll do that in a separate issue/PR.  We can try to solve the TODO for the dns subscription timer in a similar way (thought in this case we'll need to keep a list of timers per dns subscription, since they are not shared across connections, like the `ncSubscriptionKeepAliveTimer` is).